Make some changes to work with liferay 7.2.1

NaderJafari opened the pull request:
Some changes to support liferay 7.2.1 testIntegration just to get Help!
Related to this : https://gist.github.com/chrberndt/ee4c3cb4324c56301651e38a06b5b3e7#gistcomment-3345122

NaderJafari committed:
Make some changes to work with liferay 7.2.1

chrberndt commented:
Hi @naderjafari , 

Thank you very much for your pull request! While I appreciate contributions to this project, I must ask for your understanding that I won't merge your request. Other than the sources of Liferay core I prefer to not add generated sources into my project repository. While it makes testing a bit more complicated it has the benefit of giving you a clearer picture of the code you actually have to maintain. 

If you want to modify your PR accordingly and focus only on those modifications necessary to make the tests work on 7.2.1 I will be happy to consider your contribution. 

Please make sure to run the "formatSource" task on your sources in order to avoid formatting issues.

Best regards, 

Christian

naderjafari commented:
Hi @chrberndt 
I just created this PR to demonstrate to you what i changed to make the tests work on 7.2.1.
I didn't add unnecessary things.
Also i ran the "formatSource" task before committing.

**I have this error now :** 

```
java.lang.NullPointerException
	at com.chberndt.liferay.todo.list.service.persistence.test.TaskPersistenceTest.setUp(TaskPersistenceTest.java:78)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at com.liferay.portal.test.rule.TransactionalTestRule$TransactionalFrameworkMethod.access$301(TransactionalTestRule.java:183)
	at com.liferay.portal.test.rule.TransactionalTestRule$TransactionalFrameworkMethod$1.call(TransactionalTestRule.java:201)
	at com.liferay.portal.spring.transaction.BaseTransactionExecutor.execute(BaseTransactionExecutor.java:37)
	at com.liferay.portal.spring.transaction.TransactionInvokerImpl.invoke(TransactionInvokerImpl.java:39)
	at com.liferay.portal.kernel.transaction.TransactionInvokerUtil.invoke(TransactionInvokerUtil.java:28)
	at com.liferay.portal.test.rule.TransactionalTestRule$TransactionalFrameworkMethod.invokeExplosively(TransactionalTestRule.java:194)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.liferay.portal.test.rule.TransactionalTestRule$1$1.call(TransactionalTestRule.java:127)
	at com.liferay.portal.test.rule.TransactionalTestRule$1$1.call(TransactionalTestRule.java:122)
	at com.liferay.portal.spring.transaction.BaseTransactionExecutor.execute(BaseTransactionExecutor.java:37)
	at com.liferay.portal.spring.transaction.TransactionInvokerImpl.invoke(TransactionInvokerImpl.java:39)
	at com.liferay.portal.kernel.transaction.TransactionInvokerUtil.invoke(TransactionInvokerUtil.java:28)
	at com.liferay.portal.test.rule.TransactionalTestRule$1.evaluate(TransactionalTestRule.java:119)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$1.evaluate(TestExecutorRunnable.java:143)
	at com.liferay.portal.test.rule.TransactionalTestRule$1$1.call(TransactionalTestRule.java:127)
	at com.liferay.portal.test.rule.TransactionalTestRule$1$1.call(TransactionalTestRule.java:122)
	at com.liferay.portal.spring.transaction.BaseTransactionExecutor.execute(BaseTransactionExecutor.java:37)
	at com.liferay.portal.spring.transaction.TransactionInvokerImpl.invoke(TransactionInvokerImpl.java:39)
	at com.liferay.portal.kernel.transaction.TransactionInvokerUtil.invoke(TransactionInvokerUtil.java:28)
	at com.liferay.portal.test.rule.TransactionalTestRule$1.evaluate(TransactionalTestRule.java:119)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._execute(TestExecutorRunnable.java:181)
	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.run(TestExecutorRunnable.java:95)
```
If you could help me to resolve this error I'll complete test implementation on liferay 7.2.1.
also i added more details [Here](https://gist.github.com/chrberndt/ee4c3cb4324c56301651e38a06b5b3e7#gistcomment-3345122).

Thank you ❤️

chrberndt commented:
Hi @naderjafari - 

I think this PR has become obsolete with my latest updates. Updating the test-dependencies was required make the integration tests run on 7.2.1, see: https://github.com/chrberndt/liferay-todo-list/commit/ac0d5fac8fe42f6e255875cb7568e12946fe714f . 

Additionally I would recommend to upgrade your workspace to a recent version: I'm using 

```
		classpath group: "com.liferay", name: "com.liferay.gradle.plugins.workspace", version: "2.2.9"
```

in `settings.gradle`.

Thanks for your contribution & best regards, 

Christian
